scripts/deprecate_release.ts

```diff
@@ -0,0 +1,220 @@
+import { $ } from 'execa';
+import { EOL } from 'os';
+import { parseArgs } from 'util';
+import {
+  commitAllChanges,
+  configure,
+  getTagsAtCommit,
+  isCleanWorkingTree,
+  push,
+  switchToBranch,
+} from './components/git_commands.js';
+import { getOctokit, context as ghContext } from '@actions/github';
+import * as fsp from 'fs/promises';
+
+/**
+ * This script deprecates a set of package versions that were released by a single release commit.
+ *
+ * The steps that it takes are
+ * 1. Verifies that the input commit-hash is a valid release commit
+ * 2. Finds the git tags associated with that commit. These are the package versions that need to be deprecated
+ * 3. Finds the release commit _before_ the input commit-hash
+ * 4. Finds the git tags associated with the previous release commit. These are the package versions that need to be marked as "latest" (or whatever the dist-tag for the release is)
+ * 5. Creates a rollback PR that resets the .changeset directory to its state at the previous release
+ * 6. Marks the previous package versions as latest
+ * 7. Marks the current package versions as deprecated
+ */
+
+// convenience config for execa;
+const inheritIO = { stdio: 'inherit' } as const;
+
+const args = parseArgs({
+  options: {
+    'commit-hash': {
+      type: 'string',
+    },
+    'deprecation-message': {
+      type: 'string',
+    },
+  },
+});
+
+/* ====== Validation ====== */
```

Comment:
instead of Validation, Execution, etc these could be functions / classes.

Also, consider moving most of the logic to `components` and writing unit tests for them given complexity. This would also be beneficial to have this logic run in some sort of tests because we won't use it very frequently otherwise.

Reply:
I was going to make a follow up PR with a "restore_release" workflow at which point it would make sense to pull a lot of this into shared components. It might make sense to just roll that other workflow into this PR.

Reply:
> It might make sense to just roll that other workflow into this PR.

I'd be fine with that. It may force good abstractions.